src/openvic-simulation/pop/Pop.cpp

```diff
@@ -534,7 +534,7 @@ bool PopManager::load_pop_into_vector(
 	}
 
 	if (culture != nullptr && religion != nullptr && size >= 1) {
```

Comment:
```suggestion
	if (culture != nullptr && religion != nullptr && size >= 1 && size <= std::numeric_limits<Pop::pop_size_t>::max()) {
```
Can you add a max check here